addons/mrp/models/mrp_routing.py

```diff
@@ -178,16 +196,16 @@ def _skip_operation_line(self, product, never_attribute_values=False):
         return self.env['mrp.bom']._skip_for_no_variant(product, self.bom_product_template_attribute_value_ids, never_attribute_values)
 
     def _get_duration_expected(self, product, quantity, unit=False, workcenter=False):
```

Comment:
NTH. It would be great to remove this function and add a context with a depend context on computed field. The idea is to avoid code duplication and a single method